From 48edfe0770f376c9198290206a540915f52cbbd3 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Wed, 19 Nov 2014 14:24:16 +0800
Subject: [PATCH 0399/5242] MLK-11488-8 arm: imx: adjust qspi device index
 according to dtb setting

commit  41213383ca9a9eed663cd2e6a21c74a2d74d4f84 from
https://source.codeaurora.org/external/imx/linux-imx.git

When resume from DSM with Mega/Fast off, we need to restore
the right QSPI module for M4, so get the qspi index from dtb file.

Signed-off-by: Anson Huang <b20788@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-imx6.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/pm-imx6.c b/arch/arm/mach-imx/pm-imx6.c
index 439b6c9..5416f82 100644
--- a/arch/arm/mach-imx/pm-imx6.c
+++ b/arch/arm/mach-imx/pm-imx6.c
@@ -1172,7 +1172,8 @@ void __init imx6sx_pm_init(void)
 	if (np)
 		console_base = of_iomap(np, 0);
 	if (imx_src_is_m4_enabled()) {
-		np = of_find_node_by_path("/soc/aips-bus@02100000/qspi@021e4000");
+		np = of_find_compatible_node(NULL, NULL,
+				"fsl,imx6sx-qspi-m4-restore");
 		if (np)
 			qspi_base = of_iomap(np, 0);
 		WARN_ON(!qspi_base);
-- 
1.7.9.5

